src/libs/NetworkConnection.js

```diff
@@ -47,19 +49,32 @@ function setOfflineStatus(isCurrentlyOffline) {
  * internet connectivity or not. This is more reliable than the Pusher
  * `disconnected` event which takes about 10-15 seconds to emit.
  */
+function subscribeToNetInfo() {
+    // Calling NetInfo.configure (re)checks current state. We use it to force a recheck whenever we (re)subscribe
+    NetInfo.configure({
+        // By default, for web (including Electron) NetInfo uses `/` for `reachabilityUrl`
+        // When App is served locally or from Electron this would respond with OK even with no internet
+        // Using API url ensures reachability is tested over internet
+        reachabilityUrl: CONFIG.EXPENSIFY.URL_API_ROOT,
+        reachabilityTest: response => Promise.resolve(response.status === 200),
+    });
+
+    // Subscribe to the state change event via NetInfo so we can update
+    // whether a user has internet connectivity or not.
+    unsubscribeFromNetInfo = NetInfo.addEventListener((state) => {
+        Log.info('[NetworkConnection] NetInfo state', false, state);
+        setOfflineStatus(!state.isInternetReachable);
```

Comment:
Here we set the offline/online status based on `isInternetReachable` prop. I didn't find many details about it but I think this flag is set based on the result of the HTTP request.  
So when the user is online it seems right to check the flag but when the user is offline it will take some time to be set. 
I think we should do this. 

if `state.isConnected`
then  offline status = !state.isInternetReachable
Else if  `!state.isConnected` 
then offline status = false. 

When a user is not connected then we don't have to wait for this flag `state.isInternetReachable`. If it behaves the same then It's OK.

Reply:
`isInternetReachable` is not always set from the result of a request
When `state.isConnected` is `false` - `state.isInternetReachable` is also `false`

`isInternetReachable` is set from `true` to `false` when the check request fails or takes longer than 5sec.